lib/install/phases/resources.go

```diff
@@ -0,0 +1,187 @@
+			constants.FieldPhase: p.Phase.ID,
+		}),
+		Key:      opKey(p.Plan),
+		Operator: operator,
+		Server:   p.Phase.Data.Server,
+	}
+	var resources []byte
+	if p.Phase.Data != nil && p.Phase.Data.Install != nil {
+		resources = p.Phase.Data.Install.Resources
+	}
+	return &resourcesExecutor{
+		FieldLogger:    logger,
+		ExecutorParams: p,
+		resources:      resources,
+	}, nil
+}
+
+type resourcesExecutor struct {
+	// FieldLogger is used for logging
+	logrus.FieldLogger
+	// ExecutorParams is common executor params
+	fsm.ExecutorParams
+	resources []byte
+}
+
+// Execute executes the resources phase
+func (p *resourcesExecutor) Execute(ctx context.Context) error {
+	const filename = "resources.yaml"
+	p.Progress.NextStep("Creating user-supplied Kubernetes resources")
+	stateDir, err := state.GetStateDir()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	err = ioutil.WriteFile(filepath.Join(state.ShareDir(stateDir), filename),
+		p.resources, defaults.SharedReadMask)
+	if err != nil {
+		return trace.Wrap(err, "failed to write user resources on disk")
+	}
+	out, err := utils.RunInPlanetCommand(
+		ctx,
+		p.FieldLogger,
+		defaults.KubectlBin,
+		"--kubeconfig",
+		constants.PrivilegedKubeconfig,
+		"apply",
+		"-f",
+		filepath.Join(defaults.PlanetShareDir, filename),
+	)
+	if err != nil {
+		return trace.Wrap(err, "failed to create user resources: %s", out)
+	}
+	p.Info("Created user-supplied Kubernetes resources.")
+	return nil
+}
+
+// Rollback is no-op for this phase
+func (*resourcesExecutor) Rollback(ctx context.Context) error {
+	return nil
+}
+
+// PreCheck makes sure this phase is executed on a master node
+func (p *resourcesExecutor) PreCheck(ctx context.Context) error {
+	err := fsm.CheckMasterServer(p.Plan.Servers)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// PostCheck is no-op for this phase
+func (*resourcesExecutor) PostCheck(ctx context.Context) error {
+	return nil
+}
+
+// NewGravityResourcesPhase returns executor that creates Gravity resources after a successful install
+func NewGravityResourcesPhase(p fsm.ExecutorParams, operator ops.Operator) (*gravityExecutor, error) {
+	if p.Phase.Data == nil || p.Phase.Data.Install == nil {
+		return nil, trace.BadParameter("phase data is mandatory")
+	}
+	logger := &fsm.Logger{
+		FieldLogger: logrus.WithFields(logrus.Fields{
+			constants.FieldPhase: p.Phase.ID,
+		}),
+		Key:      opKey(p.Plan),
+		Operator: operator,
+		Server:   p.Phase.Data.Server,
+	}
+	factory, err := gravity.New(gravity.Config{
+		Operator: operator,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &gravityExecutor{
+		FieldLogger: logger,
+		factory:     factory,
+		resources:   p.Phase.Data.Install.Resources,
+	}, nil
+}
+
+// Execute creates the Gravity resources from the configured list
+func (r *gravityExecutor) Execute(context.Context) (err error) {
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(r.resources), defaults.DecoderBufferSize)
```

Comment:
There is a `lib/app/resources/resourcefiles.go:ForEachObject()` method that nicely abstracts this resource decoding logic and invokes a method for each decoded resource. Can we use something similar here, for gravity resources?